src/nnet3/nnet-test-utils.cc

```diff
@@ -609,6 +609,177 @@ void GenerateConfigSequenceLstmType2(
   configs->push_back(os.str());
 }
 
+// state-preserving LSTM
+void GenerateConfigSequenceStatePreservingLstm(
+    const NnetGenerationOptions &opts,
+    std::vector<std::string> *configs) {
+  std::ostringstream os;
+
+  std::vector<int32> splice_context;
+  for (int32 i = -5; i < 4; i++)
+    if (Rand() % 3 == 0)
+      splice_context.push_back(i);
+  if (splice_context.empty())
+    splice_context.push_back(0);
+
+  int32 input_dim = 10 + Rand() % 20,
+      spliced_dim = input_dim * splice_context.size(),
+      output_dim = 100 + Rand() % 200,
+      cell_dim = 40 + Rand() % 50,
+      projection_dim = std::ceil(cell_dim / (Rand() % 10 + 1));
+
+  os << "input-node name=input dim=" << input_dim << std::endl;
+  os << "input-node name=output_r_t_STATE_PREVIOUS_MINIBATCH dim="
+     << projection_dim << std::endl;
+  os << "input-node name=output_c1_t_STATE_PREVIOUS_MINIBATCH dim="
+     << cell_dim << std::endl;
+  os << "input-node name=output_c2_t_STATE_PREVIOUS_MINIBATCH dim="
+     << cell_dim << std::endl;
+
+  // Parameter Definitions W*(* replaced by - to have valid names)
```

Comment:
You could probably just call GenerateConfigSequenceLstm to generate the LSTM code.